avell-unofficial-control-center: init at 1.0.4

rkitover opened the pull request:
###### Motivation for this change

This software controls the RGB lights on the keyboard of my gaming laptop.

It's Python software that depends on pyusb, this makes it very tricky to install on the system in NixOS or in a Python virtual environment. Having a package makes this very simple.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

r-rmcgibbo commented:
Result of `nixpkgs-review pr 124234` at cb4c5a59 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>avell-unofficial-control-center</li>
    <li>python38Packages.elevate</li>
    <li>python39Packages.elevate</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/applications/misc/avell-unofficial-control-center/default.nix:18:0:
```
   |
18 |     description = "Software for controlling the keyboard backlight on some gaming laptops";
   | ^
```
</li>
    <li>warning: unused-argument

Unused argument: `stdenv`.
Near pkgs/development/python-modules/elevate/default.nix:1:8:
```
  |
1 | { lib, stdenv, fetchPypi, buildPythonPackage, fetchpatch, setuptools_scm }:
  |        ^
```
</li>
  </ul>
</details>

rkitover committed:
maintainers: add rkitover

Signed-off-by: Rafael Kitover <rkitover@gmail.com>

rkitover committed:
python3Packages.elevate: init at 0.1.3

Needed by some Python software to re-exec with super-user privileges.

Add a patch to work correctly with NixOS shell script wrappers for
python executables.

Signed-off-by: Rafael Kitover <rkitover@gmail.com>

rkitover committed:
avell-unofficial-control-center: init at 1.0.4

Software for controlling RGB keyboard lights on some gaming laptops that
use ITE Device(8291) Rev 0.03.

Signed-off-by: Rafael Kitover <rkitover@gmail.com>

SuperSandro2000 commented on pkgs/development/python-modules/elevate/default.nix:
```diff
@@ -0,0 +1,33 @@
+    sha256 = "53ad19fa1de301fb1de3f8768fb3a5894215716fd96a475690c4d0ff3b1de209";
+  };
+
+  patches = [ (fetchpatch {
+    # This is for not calling shell wrappers through Python, which fails.
+    url = "https://github.com/rkitover/elevate/commit/148b2bf698203ea39c9fe5d635ecd03cd94051af.patch";
+    sha256 = "1ky3z1jxl1g28wbwbx8qq8jgx8sa8pr8s3fdcpdhdx1blw28cv61";
+  })];
```
```suggestion
  patches = [
    (fetchpatch {
    # This is for not calling shell wrappers through Python, which fails.
    url = "https://github.com/rkitover/elevate/commit/148b2bf698203ea39c9fe5d635ecd03cd94051af.patch";
    sha256 = "1ky3z1jxl1g28wbwbx8qq8jgx8sa8pr8s3fdcpdhdx1blw28cv61";
    })
  ];
```

SuperSandro2000 commented on pkgs/development/python-modules/elevate/default.nix:
```diff
@@ -0,0 +1,33 @@
+  })];
+
+  nativeBuildInputs = [
+    setuptools_scm
```
```suggestion
    setuptools-scm
```

SuperSandro2000 commented on pkgs/development/python-modules/elevate/default.nix:
```diff
@@ -0,0 +1,33 @@
+{ lib, fetchPypi, buildPythonPackage, fetchpatch, setuptools_scm }:
```
```suggestion
{ lib, fetchPypi, buildPythonPackage, fetchpatch, setuptools-scm }:
```

SuperSandro2000 commented on pkgs/applications/misc/avell-unofficial-control-center/default.nix:
```diff
@@ -0,0 +1,25 @@
+  src = fetchFromGitHub {
+    owner = "rodgomesc";
+    repo = "avell-unofficial-control-center";
+    rev = "e32e243e31223682a95a719bc58141990eef35e6";
```
```suggestion
    # https://github.com/rodgomesc/avell-unofficial-control-center/issues/58
    rev = "e32e243e31223682a95a719bc58141990eef35e6";
```

SuperSandro2000 reviewed (commented):
some stylistic suggestions plus a comment to tag releases.

rkitover commented:
@SuperSandro2000 thank you very much, I have merged your changes.
